drivers/espi/espi_realtek_rts5912.c

```diff
@@ -679,8 +681,18 @@ static void espi_port80_isr(const struct device *dev)
 				 ESPI_PERIPHERAL_NODATA};
 	volatile struct port80_reg *const port80_reg = espi_config->port80_reg;
 
-	evt.evt_data = port80_reg->DATA;
-	espi_send_callbacks(&espi_data->callbacks, dev, evt);
+	int32_t start_time = k_uptime_get_32();
+
+	while (!(port80_reg->STS & PORT80_STS_FIFOEM)) {
```

Comment:
As implemented, you are including the time spent in the callback as part of the timeout check.  If the callback is slow, this will cause a false timeout.  You should exclude the callback time in this calculation.

Also note that it's recommended to use the 64-bit version of `k_uptime_get()`